pydantic/main.py

```diff
@@ -20,7 +21,8 @@
 from .errors import PydanticUserError
 from .fields import Field, FieldInfo, ModelPrivateAttr
 from .json import custom_pydantic_encoder, pydantic_encoder
-from .schema import default_ref_template, model_schema
+from .json_schema import DEFAULT_REF_TEMPLATE, GenerateJsonSchema
+from .json_schema_misc import JsonSchemaMisc
```

Comment:
I don't like that `json_schema_misc` is a different file than `json_schema`, but I was having circular imports issues coming from the use of `JsonSchemaMisc` in the `CoreMetadataHandler`, and the `CoreMetadataHandler` in `GenerateJsonSchema`. Putting these all in the same file may have helped, but it felt wrong to put the metadata handling stuff in an otherwise-JSON-schema-specific file.

Happy to reorganize/rename files if there are any suggestions for improvement.